Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/risk/storage/RiskResultDatabaseMigrationTest.kt

```diff
@@ -0,0 +1,117 @@
+import androidx.test.platform.app.InstrumentationRegistry
+import de.rki.coronawarnapp.risk.RiskLevelResult
+import de.rki.coronawarnapp.risk.storage.internal.RiskResultDatabase
+import de.rki.coronawarnapp.risk.storage.internal.migrations.RiskResultDatabaseMigrations
+import de.rki.coronawarnapp.risk.storage.internal.riskresults.PersistedRiskLevelResultDao
+import de.rki.coronawarnapp.server.protocols.internal.v2.RiskCalculationParametersOuterClass
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+import org.joda.time.Instant
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class RiskResultDatabaseMigrationTest {
+    private val DB_NAME = "riskresults_migration_test.db"
+
+    @get:Rule
+    val helper: MigrationTestHelper = MigrationTestHelper(
+        InstrumentationRegistry.getInstrumentation(),
+        RiskResultDatabase::class.java.canonicalName,
+        FrameworkSQLiteOpenHelperFactory()
+    )
+
+    /**
+     * Test migration to create new primary key "monotonicId" column
+     */
+    @Test
+    fun migrate1To2() {
+        helper.createDatabase(DB_NAME, 1).apply {
+            execSQL(
+                """
+                INSERT INTO "riskresults" ("id", "calculatedAt", "totalRiskLevel", "totalMinimumDistinctEncountersWithLowRisk", "totalMinimumDistinctEncountersWithHighRisk", "mostRecentDateWithLowRisk", "mostRecentDateWithHighRisk", "numberOfDaysWithLowRisk", "numberOfDaysWithHighRisk") VALUES ('72c4084a-43a9-4fcf-86d4-36103bfbd492', '2020-12-31T16:41:50.207Z', '2', '8', '1', '2020-12-29T16:41:50.038Z', '2020-12-30T16:41:50.038Z', '3', '1');
+               """.trimIndent()
+            )
+            execSQL(
+                """
+                INSERT INTO "riskresults" ("id", "calculatedAt", "totalRiskLevel", "totalMinimumDistinctEncountersWithLowRisk", "totalMinimumDistinctEncountersWithHighRisk", "mostRecentDateWithLowRisk", "mostRecentDateWithHighRisk", "numberOfDaysWithLowRisk", "numberOfDaysWithHighRisk") VALUES ('48a57f54-467b-4a0b-89c4-3c14e7ce65b5', '2020-12-31T16:41:38.663Z', '1', '0', '0', NULL, NULL, '0', '0');
+                """.trimIndent()
+            )
+            execSQL(
+                """
+                INSERT INTO "riskresults" ("id", "calculatedAt", "failureReason") VALUES ('0235fef8-4332-4a43-b7d8-f5eacb54a6ee', '2020-12-31T16:28:25.400Z', 'tracingOff');
+               """.trimIndent()
+            )
+
+            close()
+        }
+
+        // Run migration
+        helper.runMigrationsAndValidate(
+            DB_NAME,
+            2,
+            true,
+            RiskResultDatabaseMigrations.MIGRATION_1_2
+        )
+
+        val daoDb = RiskResultDatabase.Factory(
+            context = ApplicationProvider.getApplicationContext()
+        ).create(databaseName = DB_NAME)
+
+        val allEntries = runBlocking { daoDb.riskResults().allEntries().first() }
```

Comment:
`runBlockingTest`

Reply:
What would be the effective difference here?

Reply:
I think it executes the test body on an immediate dispatcher, especially if there are `delay` or `async` blocks inside.

Reply:
I'd like to leave it on `runBlocking` as we don't need any `delay` handling here and `runBlockingTest` returns `Unit` while `runBlocking` returns `T` (last type in scope) which makes the test more readable.

Reply:
Sure